pkg/driver/trace_driver.go

```diff
@@ -307,46 +237,46 @@ func (d *Driver) functionsDriver(list *list.List, announceFunctionDone *sync.Wai
 
 		previousIATSum += iat.Microseconds()
 
+		if !d.Configuration.TestMode {
+			waitForInvocations.Add(1)
+
+			go d.invokeFunction(&InvocationMetadata{
+				RootFunction:          list,
+				Phase:                 currentPhase,
+				MinuteIndex:           minuteIndex,
+				InvocationIndex:       invocationIndex,
+				SuccessCount:          &successfulInvocations,
+				FailedCount:           &failedInvocations,
+				FailedCountByMinute:   failedInvocationByMinute,
+				RecordOutputChannel:   recordOutputChannel,
+				AnnounceDoneWG:        &waitForInvocations,
+				AnnounceDoneExe:       addInvocationsToGroup,
+				ReadOpenWhiskMetadata: readOpenWhiskMetadata,
+			}, iatIndex)
```

Comment:
We should push this inside the structure. Also we need to prune the structure: we are passing minute index and invocation index just to create the name (we can do it here and pass it instead); readOpenWhiskMetadata is not used inside now.

Reply:
Done moving and pruning.